src/main/java/net/glowstone/entity/GlowPlayer.java

```diff
@@ -2122,14 +2124,111 @@ public void sendTitle(Title title) {
         session.sendAll(TitleMessage.fromTitle(title));
     }
 
+    /**
+     * Send the player a title base on a {@link Title.Builder}
+     *
+     * @param title the {@link Title.Builder} to send the player
+     */
+    public void sendTitle(Title.Builder title) {
+        sendTitle(title.build());
+    }
+
+    /**
+     * Send the player their current title
+     */
+    public void sendTitle() {
+        sendTitle(currentTitle);
+        currentTitle = new Title.Builder();
+    }
+
     @Override
     public void updateTitle(Title title) {
-        sendTitle(title); // TODO: update existing title instead of sending a new title
+        Title builtTitle = currentTitle.build();
+
+        if (title.getTitle() != null) {
```

Comment:
Right. I was just doing this because I was thinking someone might use another title, with times, or a subtitle to update the existing title... Maybe we use a 

    if (title.getTitle().equals("")) {
???